c7n/utils.py

```diff
@@ -288,3 +289,27 @@ def backoff_delays(start, stop, factor=2.0, jitter=False):
         else:
             yield cur
         cur = cur * factor
+
+
+def parse_cidrs(permissions):
+    """Process cidr ranges"""
+    cidr = []
+
+    ip_ranges = 'IpRanges' in permissions and permissions['IpRanges'] or []
```

Comment:
this should be the caller's responsibility not inline to this function, this logic is very specific to security group permissions, while this function is about parsing a cidr.